File: app/src/main/java/com/klim/us_stock/di/settings/SettingsComponent.java
package com.klim.us_stock.di.settings;

import com.klim.us_stock.ui.windows.settings.SettingsFragment;

import dagger.Subcomponent;

@SettingsScope
@Subcomponent(modules = {SettingsModule.class})
public interface SettingsComponent {

    public void inject(SettingsFragment fragment);

}

File: src/app/src/main/java/com/klim/us_stock/di/settings/SettingsComponent.kt
package com.klim.us_stock.di.settings

import com.klim.dep_in.ApplicationContextProvider
import com.klim.us_stock.ui.windows.settings.SettingsFragment
import dagger.Component

@SettingsScope
@Component(
    dependencies = [ApplicationContextProvider::class],
    modules = [SettingsModule::class]
)
interface SettingsComponent {
    fun inject(fragment: SettingsFragment?)

    class Initializer private constructor() {
        companion object {
            fun init(appComponent: ApplicationContextProvider): SettingsComponent =
                DaggerSettingsComponent.builder()
                    .applicationContextProvider(appComponent)
                    .build()
        }
    }
}